packages/truffle-core/lib/testing/soliditytest.js

```diff
@@ -94,81 +110,163 @@ var SolidityTest = {
+            path.join(__dirname, "AssertUint.sol"),
+            path.join(__dirname, "AssertUintArray.sol"),
+            "truffle/DeployedAddresses.sol", // generated by deployed.js
+            path.join(__dirname, SafeSend)
+          ],
+          quiet: true
+        }),
+        (err, contracts) => {
+          if (err) return callback(err);
+
+          // Set network values.
+          Object.keys(contracts).forEach(name => {
+            contracts[name].network_id = runner.config.network_id;
+            contracts[name].default_network = runner.config.default_network;
+          });
+
+          runner.config.artifactor
+            .saveAll(contracts, runner.config.contracts_build_directory)
+            .then(() => {
+              callback();
+            })
+            .catch(callback);
+        }
+      );
     });
   },
 
-  deployTestDependencies: function(abstraction, dependency_paths, runner, callback) {
-    var deployer = new Deployer(runner.config.with({
-      logger: { log: function() {} }
-    }));
+  deployTestDependencies(abstraction, dependency_paths, runner, callback) {
+    const deployer = new Deployer(
+      runner.config.with({
+        logger: { log() {} }
+      })
+    );
 
-    var Assert = runner.config.resolver.require("truffle/Assert.sol");
-    var DeployedAddresses = runner.config.resolver.require("truffle/DeployedAddresses.sol");
-    var SafeSend = runner.config.resolver.require("SafeSend.sol");
+    const Assert = runner.config.resolver.require("truffle/Assert.sol");
+    const AssertAddress = runner.config.resolver.require(
+      "truffle/AssertAddress.sol"
+    );
+    const AssertAddressArray = runner.config.resolver.require(
+      "truffle/AssertAddressArray.sol"
+    );
+    //const AssertAddressPayableArray = runner.config.resolver.require(
+    // "truffle/AssertAddressPayableArray.sol"
+    //);
+    const AssertBalance = runner.config.resolver.require(
+      "truffle/AssertBalance.sol"
+    );
+    const AssertBool = runner.config.resolver.require("truffle/AssertBool.sol");
+    const AssertBytes32 = runner.config.resolver.require(
+      "truffle/AssertBytes32.sol"
+    );
+    const AssertBytes32Array = runner.config.resolver.require(
+      "truffle/AssertBytes32Array.sol"
+    );
+    const AssertGeneral = runner.config.resolver.require(
+      "truffle/AssertGeneral.sol"
+    );
+    const AssertInt = runner.config.resolver.require("truffle/AssertInt.sol");
+    const AssertIntArray = runner.config.resolver.require(
+      "truffle/AssertIntArray.sol"
+    );
+    const AssertString = runner.config.resolver.require(
+      "truffle/AssertString.sol"
+    );
+    const AssertUint = runner.config.resolver.require("truffle/AssertUint.sol");
+    const AssertUintArray = runner.config.resolver.require(
+      "truffle/AssertUintArray.sol"
+    );
+    const DeployedAddresses = runner.config.resolver.require(
+      "truffle/DeployedAddresses.sol"
+    );
```

Comment:
Can we reduce the boilerplate here a bit?

Something like:
```javascript
const assertLibraries = [
  "AssertAddress",
  "AssertAddressArray",
  "AssertBalance",
  "AssertBool",
  "AssertBytes32",
  "AssertGeneral",
  "AssertInt",
  "AssertIntArray",
  "AssertString",
  "AssertUint",
  "AssertUintArray"
];

const assertAbstractions = assertLibraries.map(
  (name) => runner.config.resolver.require(`truffle/${name}.sol`)
);
```

Then, below, we can change `deployer` to use `await` syntax:

```javascript
await deployer.deploy(Assert);
for (let abstraction of assertAbstractions) {
  await deployer.deploy(abstraction);
}
/* ... continue into dependency_paths stuff ... */
```

What do you think?